middleware/kubernetes/kubernetes.go

```diff
@@ -0,0 +1,213 @@
+// Package kubernetes provides the kubernetes backend.
+package kubernetes
+
+import (
+    "fmt"
+	"strings"
+	"time"
+
+	"github.com/miekg/coredns/middleware"
+	"github.com/miekg/coredns/middleware/kubernetes/msg"
+	k8sc "github.com/miekg/coredns/middleware/kubernetes/k8sclient"
+	"github.com/miekg/coredns/middleware/proxy"
+//	"github.com/miekg/coredns/middleware/singleflight"
+
+    "github.com/miekg/dns"
+	"golang.org/x/net/context"
+)
+
+type Kubernetes struct {
+	Next       middleware.Handler
+	Zones      []string
+	Proxy      proxy.Proxy // Proxy for looking up names during the resolution process
+	Ctx        context.Context
+//	Inflight   *singleflight.Group
+    APIConn    *k8sc.K8sConnector
+}
+
+
+func (g Kubernetes) getZoneForName(name string) (string, []string) {
+    var zone string
+    var segments []string
+
+    for _, z := range g.Zones {
+        if dns.IsSubDomain(z, name) {
+            zone = z 
+    
+            segments = dns.SplitDomainName(name)
+            segments = segments[:len(segments) - dns.CountLabel(zone)]
```

Comment:
this keeps overwriting segments?

What does this function do, function comment would help here :-)